lib/bios/menus.robot

```diff
@@ -100,6 +100,62 @@ Get Boot Menu Construction
         END
     END
     RETURN    ${construction}
+Search For The Option
+    [Documentation]
+    ...    Reads the serial output in search for the first occurance of
+    ...    ${option} when it fidns it, the function returns the
+    ...    qantity of ${ARROW_DOWN} presses required to reach that ${option}
+    ...
+    ...    === Requirements ===
+    ...    - Boot menu has to be entered using ``Enter Boot Menu Tianocore``
+    ...    - (?)The serial must not have been read after entering the boot menu
+    ...
+    ...    === Arguments ===
+    ...    ``${option}``: ``string`` The first line of the option you want to find.
+    ...    In case options are split into multiple lines make sure to put only
+    ...    the first line of the option as argument.
+    ...
+    ...    === Return Value ===
+    ...    - ``int`` - The qantity of ${ARROW_DOWN} presses required to reach that ${option}
+    ...    - ``-1`` - If the  ${option} has not been found.
+    ...
+    ...    === Effects ===
+    ...    - The submenu is read from the serial buffer
+    [Arguments]    ${option}
+    ${menu}=    Read From Terminal Until    Exit
+    # Lines to strip:
+    #    TOP:
+    #    Please select boot device:
+    #    BOTTOM
+    #    ^ and v to move selection
+    #    ENTER to select boot device
+    #    ESC to exit
```

Comment:
This is valid for boot options, but not for setup menu. This may impact numbers used in the function (e.g. how many lines to cut from top and bottom, or how many entries are displayed on one screen), have you checked that they are correct?